model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/stencils/compute_edge_diagnostics_for_dycore_and_update_vn.py

```diff
@@ -0,0 +1,678 @@
+    _compute_horizontal_gradient_of_exner_pressure_for_multiple_levels,
+)
+from icon4py.model.atmosphere.dycore.stencils.compute_horizontal_gradient_of_exner_pressure_for_nonflat_coordinates import (
+    _compute_horizontal_gradient_of_exner_pressure_for_nonflat_coordinates,
+)
+from icon4py.model.atmosphere.dycore.stencils.mo_math_gradients_grad_green_gauss_cell_dsl import (
+    _mo_math_gradients_grad_green_gauss_cell_dsl,
+)
+from icon4py.model.common import dimension as dims, field_type_aliases as fa, type_alias as ta
+from icon4py.model.common.type_alias import vpfloat, wpfloat
+
+
+@gtx.field_operator
+def _compute_theta_rho_face_values_and_pressure_gradient_and_update_vn_in_predictor_step(
+    rho_at_edges_on_model_levels: fa.EdgeKField[ta.wpfloat],
+    theta_v_at_edges_on_model_levels: fa.EdgeKField[ta.wpfloat],
+    horizontal_pressure_gradient: fa.EdgeKField[ta.vpfloat],
+    next_vn: fa.EdgeKField[ta.wpfloat],
+    current_vn: fa.EdgeKField[ta.wpfloat],
+    tangential_wind: fa.EdgeKField[ta.vpfloat],
+    reference_rho_at_edges_on_model_levels: fa.EdgeKField[ta.vpfloat],
+    reference_theta_at_edges_on_model_levels: fa.EdgeKField[ta.vpfloat],
+    perturbed_rho: fa.CellKField[ta.vpfloat],
+    perturbed_theta_v: fa.CellKField[ta.vpfloat],
+    temporal_extrapolation_of_perturbed_exner: fa.CellKField[ta.vpfloat],
+    ddz_temporal_extrapolation_of_perturbed_exner_on_model_levels: fa.CellKField[ta.vpfloat],
+    d2dz2_temporal_extrapolation_of_perturbed_exner_on_model_levels: fa.CellKField[ta.vpfloat],
+    hydrostatic_correction_on_lowest_level: fa.EdgeField[ta.wpfloat],
+    predictor_normal_wind_advective_tendency: fa.EdgeKField[ta.vpfloat],
+    normal_wind_tendency_due_to_physics_process: fa.EdgeKField[ta.vpfloat],
+    normal_wind_iau_increments: fa.EdgeKField[ta.vpfloat],
+    geofac_grg_x: gtx.Field[[dims.CellDim, dims.C2E2CODim], ta.wpfloat],
+    geofac_grg_y: gtx.Field[[dims.CellDim, dims.C2E2CODim], ta.wpfloat],
+    pos_on_tplane_e_x: gtx.Field[[dims.ECDim], ta.wpfloat],
+    pos_on_tplane_e_y: gtx.Field[[dims.ECDim], ta.wpfloat],
+    primal_normal_cell_x: gtx.Field[[dims.ECDim], ta.wpfloat],
+    dual_normal_cell_x: gtx.Field[[dims.ECDim], ta.wpfloat],
+    primal_normal_cell_y: gtx.Field[[dims.ECDim], ta.wpfloat],
+    dual_normal_cell_y: gtx.Field[[dims.ECDim], ta.wpfloat],
+    ddxn_z_full: fa.EdgeKField[ta.vpfloat],
+    c_lin_e: gtx.Field[[dims.EdgeDim, dims.E2CDim], ta.wpfloat],
+    ikoffset: gtx.Field[[dims.ECDim, dims.KDim], gtx.int32],
+    zdiff_gradp: gtx.Field[[dims.ECDim, dims.KDim], ta.vpfloat],
+    ipeidx_dsl: fa.EdgeKField[bool],
+    pg_exdist: fa.EdgeKField[ta.vpfloat],
+    inv_dual_edge_length: fa.EdgeField[ta.wpfloat],
+    dtime: ta.wpfloat,
+    cpd: ta.wpfloat,
+    iau_wgt_dyn: ta.wpfloat,
+    is_iau_active: bool,
+    limited_area: bool,
+    iadv_rhotheta: gtx.int32,
+    igradp_method: gtx.int32,
+    MIURA_advection_type: gtx.int32,
+    TAYLOR_HYDRO_gradp_method: gtx.int32,
```

Comment:
Could we not use the `FrozenNamespace` trick here to not have to pass these constant values (see `saturation_adjustment.py`)

Reply:
this makes a lot of sense